### ISSUE TITLE
cmd/go: go mod tidy installs unreleased and gone package

### ISSUE THREAD

**Issue by vmihailenco** [golang/go#51888]
Using Go 1.8 with this program:

```go
// main.go
package main

import (
	"fmt"

	"github.com/uptrace/go-clickhouse/ch"
)

func main() {
	db := ch.Connect(
		// clickhouse://<user>:<password>@<host>:<port>/<database>?sslmode=disable
		ch.WithDSN("clickhouse://localhost:9000/default?sslmode=disable"),
	)
	fmt.Println(db.Ping())
}
```

Then I run:

```shell
go clean -modcache
go mod init example
go mod tidy

go: downloading github.com/uptrace/go-clickhouse/ch v0.0.0-20220308124651-82a5d8d72ef7
go: downloading github.com/uptrace/go-clickhouse v0.2.2
```

The problem is that Go installs `github.com/uptrace/go-clickhouse/ch` package that was never properly released but was installed once using Git revision. `github.com/uptrace/go-clickhouse/ch/go.mod` is long gone from the repository. I want `github.com/uptrace/go-clickhouse v0.2.2` to be installed instead.

How can I completely remove `github.com/uptrace/go-clickhouse/ch` from the cache?

**Comment by seankhliao** [comment]
uptrace/go-clickhouse will need to use `retract` to fix it.

**Comment by seankhliao** [comment]
as an user, you can use `exclude` while they fix it

**Comment by vmihailenco** [comment]
Please read the issue. There is no go.mod file and there is no version to retract. The package `github.com/uptrace/go-clickhouse/ch` is gone and replaced by `github.com/uptrace/go-clickhouse`.

**Comment by seankhliao** [comment]
There was publicly available commit and it is cached. It needs to be retracted.

**Comment by vmihailenco** [comment]
> It needs to be retracted.

Or deleted. That is why I've created this issue.

To retract it, I need to update go.mod file for the package `github.com/uptrace/go-clickhouse/ch`. Such package does not exist any more and go.mod is gone. The package was replaced with `github.com/uptrace/go-clickhouse`.

I could restore go.mod for `github.com/uptrace/go-clickhouse/ch` to retract the old version. But then Go will cache a new version for that package and I will have the same problem. I want the package `github.com/uptrace/go-clickhouse/ch` to be completely gone.

Is there a way to manually remove https://proxy.golang.org/github.com/uptrace/go-clickhouse/ch/@v/v0.0.0-20220308124651-82a5d8d72ef7.mod?
